### PR TITLE
Context precision with ground truth

### PR DESCRIPTION
**Pull request by shahules786**
#365 fixes

### CODE REVIEW

**Comment by shahules786** [comment]
@yuukidach Hey, I would love to know your thoughts on my approach here.

**Comment by yuukidach** [comment]
@shahules786 It's really not easy to merge two PRs. I think we can directly adopt yours. I'll closed my PR.

1. In this PR, there is another metric for dataset with out gt. This makes prompt tuning more convenient
2. Since the prompt are simpler ans shorter, the running time can also be faster.

And for the evaluation of context_relevancy, I agree with your idea. There is a high probability that using GT/Answer can also improve the effect.

BTW, when tuning the prompt, I found that a suitable prompt is still very important in the evaluation process. Will Ragas consider let users to use our own prompts or provide a module that automatically tuning prompt like https://arxiv.org/abs/2309.03409

File: src/ragas/metrics/_context_precision.py [hunk 1]
```diff
@@ -51,26 +69,40 @@ class ContextPrecision(MetricWithLLM):
...
     name: str = "context_precision"  # type: ignore
-    evaluation_mode: EvaluationMode = EvaluationMode.qc  # type: ignore
+    evaluation_mode: EvaluationMode = EvaluationMode.qcg  # type: ignore
     batch_size: int = 15
 
+    def get_dataset_attributes(self, dataset: Dataset):
+        answer = "ground_truths"
+        if answer not in dataset.features.keys():
+            logging.warning(
+                "Using context precision without ground truth will be soon depreciated. Use context_utilization instead"
```
Comment (jjmachan) [inline src/ragas/metrics/_context_precision.py:79]:
```suggestion
                "Using 'context_precision' without ground truth will be soon depreciated. Use 'context_utilization' instead"
```

File: src/ragas/validation.py [hunk 2]
```diff
@@ -60,8 +61,15 @@ def validate_evaluation_modes(ds: Dataset, metrics: list[Metric]):
         required_columns = set(EVALMODE_TO_COLUMNS[m.evaluation_mode])
         available_columns = set(ds.features.keys())
         if not required_columns.issubset(available_columns):
+            extra_msg = ""
+            if (
+                isinstance(m, ContextPrecision)
+                and "ground_truths" not in available_columns
+            ):
+                extra_msg = "Looks like you're trying to use context precision without ground_truths. Please use context_utilization"
```
Comment (jjmachan) [inline src/ragas/validation.py:69]:
```suggestion
                extra_msg = "Looks like you're trying to use 'context_precision' without ground_truths. Please use consider using  `context_utilization' instead."
```

**Review by jjmachan — APPROVED** [review]
awesome! this looks good

**Pull request merged**